### PR TITLE
Support eip range

### PR DESCRIPTION
**Pull request by k0ngk0ng**
details:
1. remove iptables forward rules for eip, it's useless.
1. change the pattern of eip.spec.address in CRD to support 'ip1~ip2'
   now we support 3 kinds of address syntax
    ```
    - ip        e.g.  192.168.0.1
    - ip/net    e.g.  192.168.0.0/24
    - ip1~ip2   e.g.  192.168.0.1~192.168.0.10
   ```
 
 I did some testing for this. pls review it thanks.

### CODE REVIEW

**Comment by zheng1** [comment]
👍 Great work! I'll review this PR soon.

File: pkg/constant/constants.go [hunk 1]
```diff
@@ -4,4 +4,5 @@ const (
 	FinalizerName     string = "finalizer.lb.kubesphere.io/v1alpha1"
 	NodeFinalizerName string = "finalizer.lb.kubesphere.io"
 	IPAMFinalizerName string = "finalizer.ipam.kubesphere.io/v1alpha1"
+	EipRangeSeparator string = "~"
```
Comment (zheng1) [inline pkg/constant/constants.go:7]: Why not use`-` to separate? just like MetalLB

File: pkg/ipam/datastore.go [hunk 2]
```diff
@@ -46,6 +52,110 @@ func (c *CIDRResource) IsFull() bool {
...
+		return c.CIDR.Contains(ip)
+	}
+
+	start := binary.BigEndian.Uint32(c.FirstIP.To4())
+	end := binary.BigEndian.Uint32(c.LastIP.To4())
+	value := binary.BigEndian.Uint32(ip.To4())
+	return start <= value && value <= end
+}
+
+func (c *CIDRResource) IntersectsWith(value *CIDRResource) bool {
```
Comment (zheng1) [inline pkg/ipam/datastore.go:70]: It looks like the implementation of MetalLB is more clear?
https://sourcegraph.com/github.com/metallb/metallb@b6678ebb6cb7b0ec591e88de9b34808d59212b1e/-/blob/internal/config/config.go#L455:6

File: pkg/util/ip.go [hunk 3]
```diff
@@ -42,7 +46,34 @@ func ToCommonString(ip string, prefix uint32) string {
 	return fmt.Sprintf("%s/%d", ip, prefix)
 }
 
-func GetValidAddressCount(cidr string) int {
+func GetAddressRange(addr string) (net.IP, net.IP, *net.IPNet, error) {
```
Comment (zheng1) [inline pkg/util/ip.go:49]: I think we can save a lot of code if we refer to the implementation of MetalLB
https://sourcegraph.com/github.com/metallb/metallb@b6678ebb6cb7b0ec591e88de9b34808d59212b1e/-/blob/internal/config/config.go#L422:6

For example: convert IP range to CIDR

File: pkg/constant/constants.go (hunk 1 above)
Comment (k0ngk0ng) [reply pkg/constant/constants.go:7]: OK, I'll change it.

File: pkg/ipam/datastore.go (hunk 2 above)
Comment (k0ngk0ng) [reply pkg/ipam/datastore.go:70]: MetalLB uses the package `mikioh/ipaddr` to convert the IP range to multiple `net.IPNet`.  
And then compare these IPNets to the targets. I think this code is not very efficient.
For example, we set `192.168.0.1-192.168.0.10` to the address, and then the function `ipaddr.Summarize(start, end)` will retrun 5 IPNets 
```
192.168.0.1/32
192.168.0.2/31
192.168.0.4/30
192.168.0.8/31
192.168.0.10/32
```
In my way, I only compare 2 firstIP and 2 lastIP. 
But if we use MetalLB's implementation, the code may looks more simpler.  😎

File: pkg/util/ip.go (hunk 3 above)
Comment (k0ngk0ng) [reply pkg/util/ip.go:49]: OK, I can try

**Comment by duanjiong** [comment]
Should also update doc about eip usage.

**Comment by zheng1** [comment]
LGTM. Please update the doc about how to use it.

**Comment by k0ngk0ng** [comment]
OK, the PR is https://github.com/kubesphere/porter/pull/93

**Pull request merged**